src/Microsoft.AspNet.Authentication/ClaimsTransformationAuthenticationHandler.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
```

Comment:
>>> Unfortunately understanding middlewares is pretty much required in ASPNET5 now.
That is actually a shame.

But I have to disagree with this fix, it is way too complicated.
For a user that wants to use a single OIDC provider all they should need to know about the middleware is app.UseOIDC(clientId, authority).  Nothing else.

I am currently fixing things so they don't even have to think about 'CookieMiddleWare'.
This change moves us away from the simple model and forces the user to know about types and code not associated with what they are trying to do.